cloudpickle_generators/_core.c

```diff
@@ -312,14 +323,14 @@ restore_frame(PyObject* UNUSED(self), PyObject* args, PyObject* kwargs) {
             return NULL;
         }
 
-        frame->f_exc_type = PyTuple_GET_ITEM(exc_info, 0);
-        Py_INCREF(frame->f_exc_type);
+        EXC_TYPE_REF(frame) = PyTuple_GET_ITEM(exc_info, 0);
+        Py_INCREF(EXC_TYPE_REF(frame));
 
-        frame->f_exc_value = PyTuple_GET_ITEM(exc_info, 0);
-        Py_INCREF(frame->f_exc_value);
+        EXC_VALUE_REF(frame) = PyTuple_GET_ITEM(exc_info, 1);
+        Py_INCREF(EXC_VALUE_REF(frame));
 
-        frame->f_exc_traceback = PyTuple_GET_ITEM(exc_info, 0);
-        Py_INCREF(frame->f_exc_traceback);
+        EXC_TRACEBACK_REF(frame) = PyTuple_GET_ITEM(exc_info, 2);
```

Comment:
Thanks for fixing this. I am trying to figure out why this didn't just fail. Probably some missing test coverage, or maybe the exception normalization code handled this in a weird way.